### PR TITLE
Context-Sensitive Completions for Macro and Builtin Command Prefixes

### PR DESCRIPTION
**Pull request by robottwo**
Hi there! I'd like to propose this enhancement to improve the user experience with intelligent tab completion for special prefixes.

### What does this PR do?

This PR adds context-sensitive completions for `#/` (macro) and `#!` (builtin command) prefixes, making the shell more intuitive and efficient to use.

### Key Features

**Smart Prefix Recognition:**
- **`#/` prefix**: Automatically suggests available macros when typing `#/` followed by tab
- **`#!` prefix**: Intelligently completes to built-in commands when using the `#!` prefix

**Enhanced User Experience:**
- **Multi-word support**: Properly handles completions with spaces and word boundaries
- **Clean cycling**: No text retention artifacts when cycling through different completion lengths
- **Graceful fallback**: Falls back to path completion when no macro/builtin matches are found

### Examples

```bash
#/m<TAB>     → Shows available macros like #/macro1, #/macro2, #/macro3
#!n<TAB>     → Suggests builtin commands like #!new
```

### Technical Implementation

- Enhanced completion provider with context-aware logic
- Comprehensive test coverage for various completion scenarios
- Improved word boundary detection for complex completions
- Fixed text retention issues during completion cycling

### Testing

All changes are thoroughly tested with:
- Unit tests for completion logic
- Integration tests for real-world scenarios
- Edge case handling for unusual input patterns

This enhancement maintains backward compatibility while significantly improving the user experience for power users who frequently use macros and builtin commands.

Thank you for considering this contribution! I'm happy to address any feedback or suggestions you might have.

### CODE REVIEW

**Comment by atinylittleshell** [comment]
@robottwo thanks a lot - this sounds very useful. Do you think you can attach a gif showing how this works in practice? I'll review more deeply once we merge the other PR.

**Comment by robottwo** [comment]
demo

**Comment by atinylittleshell** [comment]
there's a merge conflict - can you resolve that when you get a chance?

**Comment by robottwo** [comment]
> there's a merge conflict - can you resolve that when you get a chance?

Resolved

File: gsh
[the remark this answers is not in the document]
Comment (robottwo) [reply gsh:1]: Fixed

File: demo.cast
[the remark this answers is not in the document]
Comment (robottwo) [reply demo.cast:1]: Fixed

File: internal/agent/tools/ctrl_c_removal_test.go
[the remark this answers is not in the document]
Comment (robottwo) [reply internal/agent/tools/ctrl_c_removal_test.go:1]: Added more tests for better code coverage.

File: demo.gif
[the remark this answers is not in the document]
Comment (robottwo) [reply demo.gif:1]: fixed

File: pkg/gline/app.go [hunk 1]
```diff
@@ -198,6 +203,22 @@ func (m appModel) View() string {
...
+	// Add completion box if active
+	completionBox := m.textInput.CompletionBoxView()
+	if completionBox != "" {
+		s += "\n"
+		s += m.completionStyle.Render(completionBox)
+	}
+
+	// Add help box if active
+	helpBox := m.textInput.HelpBoxView()
+	if helpBox != "" {
```
[the remark this answers is not in the document]
Comment (robottwo) [reply pkg/gline/app.go:216]: Addressed, I think
